catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -0,0 +1,261 @@
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define*/
+define([
+    'icanhaz',
+    'marionette',
+    'underscore',
+    'jquery',
+    'q',
+    'js/view/ModalRegistry.view.js',
+    'js/view/EmptyView.js',
+    'wreqr',
+    'js/view/Utils.js',
+    'text!templates/deleteRegistryModal.handlebars',
+    'text!templates/deleteRegistry.handlebars',
+    'text!templates/registryPage.handlebars',
+    'text!templates/registryList.handlebars',
+    'text!templates/registryRow.handlebars',
+    'poller',
+    'js/model/RemoteStatus.js',
+    'js/model/Status.js'
+],
+function (ich,Marionette,_,$,Q,ModalRegistry,EmptyView,wreqr,Utils,deleteRegistryModal,deleteRegistry,registryPage,registryList,registryRow,poller,RemoteStatus,Status) {
+
+    var RegistryView ={};
+
+    ich.addTemplate('deleteRegistryModal', deleteRegistryModal);
+    ich.addTemplate('deleteRegistry', deleteRegistry);
+    ich.addTemplate('registryPage', registryPage);
+    ich.addTemplate('registryRow', registryRow);
+    ich.addTemplate('registryList', registryList);
+
+    RegistryView.RegistryRow = Marionette.Layout.extend({
+        template: "registryRow",
+        tagName: "tr",
+        className: "highlight-on-hover",
+        regions: {
+            editModal: '.modal-container'
+        },
+        events: {
+            'click td' : 'editRegistry'
+        },
+        initialize: function(){
+            _.bindAll(this);
+            this.listenTo(this.model, 'change', this.render);
+            if(this.model.get("registryConfiguration").at(0)){
+               this.setupPollers(this.model.get("registryConfiguration").at(0));
+            }
+
+        },
+        editRegistry: function(evt){
+            evt.stopPropagation();
+            var service = this.model;
+            wreqr.vent.trigger('editRegistry', service);
+        },
+        updateStatus: function(status) {
+            this.model.set('available', status.get('value'));
+            this.render();
+        },
+        updateRemoteStatus: function(remoteStatus) {
+            if(remoteStatus.get('value')){
+                this.model.set('remoteName', remoteStatus.get('value').remoteName);
+                this.render();
+            }
+        },
+        setupPollers: function(config){
+            var pid = config.id;
+            var view = this;
+            this.statusModel = new Status.Model(pid);
+            this.remoteStatusModel = new RemoteStatus.Model(pid);
+            this.statusModel.on('sync', function() {
```

Comment:
For the future, you can just do 'this.listenTo(this.statusModel, 'sync', this.updateStatus)' and then it will automatically stop listening when the view closes.